Added --local argument reference

gargol opened the pull request:
no issue

- Had bumped into lack of docs when trying to install Ghost locally on a specific version
- Another person bumped into the same issue here: https://github.com/TryGhost/Ghost/issues/10813#issuecomment-505750753
- The option is speced in a test here: https://github.com/TryGhost/Ghost-CLI/blob/4e37d56b4c7c8ec61f5c82a30e2b768edc088ae2/test/unit/commands/install-spec.js#L109-L137

ErisDS commented:
I think it got missed/dropped again because it isn't included in the output from `ghost install --help`. I went through this and tried to update and improve this output along with the last docs overhaul I did. Will take a look and see what can be done.

Ultimately this is a point of confusion because the install command excepts an argument, which can be either the word "local" as in `ghost install local` or a version as in `ghost install 2.25.0` but it won't accept 2 arguments e.g. `ghost install local 2.25.0` and therefore you have to flip to using a flag `ghost install 2.25.0 --local`. I think this is a design flaw in the CLI.

For docs, I think it makes sense to put this command example in the "arguments" section as an example of how to use `ghost install [version]` rather than in the flags section, as this flag _only_ works in this one case.

gargol commented:
@kymellis @ErisDS thanks for insightful feedback. Moved the command to "arguments" section. Added: `# Note: --local flag only works with [version] parameter` part, not sure if it is too much or fine providing more context?

>  I think this is a design flaw in the CLI.

For me, the confusion in this command is the special treatment of `local` in version parameters. Personally would be for just having `--local` flag, as that's more explicit.

JohnONolan commented:
I don't know if we have this anywhere in the tone/style docs, but `#note: special use-case` is definitely an antipattern (same concept as putting alert boxes on the page)

We should either have a product that doesn't need a special note, or docs which have 1 clear way of doing something, ideally both

gargol commented:
Hm... Valid point. To follow a pattern we have in other commands, we could modify how `--local` command is framed in this doc and move it into "options" section with the following annotations:
```
# Install locally with the latest version
ghost install --local

# Install locally with a specific version
ghost install --local [version]
```

This would be analogous to how `--dir` and `--zip` flags accept paths:
```
# Select the directory to install Ghost in
ghost install --dir path/to/dir

# Install Ghost from a specific zip (useful for testing or custom builds)
ghost install --zip path/to/file.zip
```

Don't have a good idea on how to avoid ambiguity with `ghost install local` though and we also "hide" `ghost install [version] --local` that is still available.

gargol commented:
@JohnONolan @kymellis @ErisDS ping
